src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/OffenderService.kt

```diff
@@ -489,7 +490,8 @@ class OffenderService(
                   accessResponse.deserializeTo<UserOffenderAccess>()
                   return PersonInfoResult.Success.Restricted(crn, offender.otherIds.nomsNumber)
                 } catch (exception: Exception) {
-                  accessResponse.throwException()
+                  Sentry.captureException(RuntimeException("LAO calls returns but failed to marshal the response", exception))
+                  return PersonInfoResult.Success.Restricted(crn, offender.otherIds.nomsNumber)
```

Comment:
@pezholio this is worth making you aware of as even though CAS1 have manual assignments it could theoretically affect CAS1 users too. 

I think failing safely and returning a restricted person should there be _any_ problems getting the offender details on an LAO check is better than failing badly and breaking wider functionality.

The likelihood? We have only one recorded case so far (maybe we missed earlier cases) where the response isn't something we can deserialise. This means it could happen again and is worth closing off in my view.

The risk? We may return offender info presented as an LAO when in fact they aren't LAO but there was a problem accessing their info - this could be confusing to staff.

We could go further and change the type of response to something like `PersonInfoResult.Failed` and handle that explicitly into the frontend. We could say "Limited offender details" instead of "limited access offender". Whether we do that now or wait to see if this happens via Sentry alerts is a question.

Reply:
Yep, I'm happy with that. Seems like a safer option anyway